### PR TITLE
Refactor runtime to spin up http server at startup instead of per invocation.

### PR DESCRIPTION
**Pull request by taeold**
Today, emulated HTTP function is invocation works as follows:

1) Functions Emulator (FE) receives HTTP request from client
2) FE sends an IPC to a runtime process assigned to the triggered HTTP function.
3) The runtime process spins up a new Express server and begins listening to the socket assigned via `PORT`.
4) FE spins while sending `__/health` request to the port assigned to the worker process. When the socket is ready, forwards the request to the assigned port.

It seems to me a bit wasteful to spin up an express server for every invocation of the function, so in this PR we spin up the HTTP server at startup. One downside of this change is that function timeout is no longer respected for HTTP function. There isn't a quick way to fix that in small chunk - we'll tackle that problem later in https://github.com/firebase/firebase-tools/pull/4867.

### CODE REVIEW

**Comment by taeold** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4861** <a href="https://app.graphite.dev/github/pr/firebase/firebase-tools/4861" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/firebase/firebase-tools/4861?utm_source=stack-comment).

**Review by TheIronDev — APPROVED** [review]

**Comment by taeold** [comment]
I'm blocking merging the PR until https://github.com/firebase/firebase-tools/issues/4807 is closed. It seems like Emulator refactoring is affecting many developers.

File: src/emulator/functionsEmulatorRuntime.ts
Comment (taeold) [reply src/emulator/functionsEmulatorRuntime.ts:1015]: Good q - it's set at `initializeRuntime` which is executed when the process starts.

**Comment by taeold** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/firebase/firebase-tools/4861?utm_source=stack-comment).


Job Reference: fQxj0krvewveagu4c7Lc

**Comment by taeold** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/firebase/firebase-tools/4861?utm_source=stack-comment).


Job Reference: KexZc1Ygnm1Oyjv7jY5k

**Pull request merged**
